### PR TITLE
first version of database schema

### PR DESCRIPTION
**Pull request by Iicytower**
I added the first version database Schema,

### CODE REVIEW

File: backend/src/database/interfaces.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { Document } from "mongoose";
+
+export type UserRole = 'Admin' | 'RegisteredUser' | 'User';
```
Comment (Tubka) [inline backend/src/database/interfaces.ts]: We should rename RegisterUser to User - user is our user in the system, but the unregistered user should be name UnregisteredUser?

File: backend/src/database/interfaces.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+export enum role {
+  Admin= 'Admin',
+  RegisteredUser = 'RegisteredUser',
+  User = 'User',
+}
```
Comment (Tubka) [inline backend/src/database/interfaces.ts]: You created an enum, but in line 3 you used strings. You should use the enum role in line 3.

File: backend/src/database/models/Errors.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+
+import { model, Schema } from "mongoose"
+import { IErrors, Devices } from "../interfaces";
```
Comment (Tubka) [inline backend/src/database/models/Errors.ts]: Where do you use the Devices interface?

File: backend/src/database/models/Links.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  },
+  destinationLink: {
+    required: true,
+    type: String,
+  },
+  verified: {
+    required: true,
+    type: Boolean,
+    default: false,
+  },
```
Comment (Tubka) [inline backend/src/database/models/Links.ts]: Verified as boolean? We have a three-state: safe, unsafe, and unverified.

File: backend/src/database/models/Links.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  description: {
+    required: true,
+    type: String,
+    default: '',
+  },
+  group: {
+    required: true,
+    type: String,
+    default: 'main',
+  },
```
Comment (Tubka) [inline backend/src/database/models/Links.ts]: What is the group? This is a links group? Why is required? Always we will create groups, even to a single link?

File: backend/src/database/models/User.ts [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import { model, Schema } from "mongoose"
+import { IUser, Devices } from "../interfaces";
```
Comment (Tubka) [inline backend/src/database/models/User.ts]: Where do you use the Devices interface?

File: backend/src/database/database.ts [hunk 7]
```diff
@@ -17,4 +17,8 @@ export async function connectDB () {
   } catch (error) {
     console.error('Error with connecting to database: \n', error);
   }
-}
\ No newline at end of file
+}
+
+export * from './models/Errors';
```
Comment (Iicytower) [inline backend/src/database/database.ts]: should be Error and Link, without `s`

Comment (Iicytower) [reply backend/src/database/database.ts]: fixed

File: backend/src/database/models/Links.ts (hunk 5 above)
Comment (Iicytower) [reply backend/src/database/models/Links.ts]: you're right the group should be not required and I deleted default value

**Pull request closed without merging**